### ISSUE TITLE
IDE0009 false positive on local functions

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#38043]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/content/problem/695869/ide0009-false-positive-on-local-functions.html)._

---
[regression] [worked-in:16.3.0 Preview 2.0]
On some local functions IDE0009 is triggered e.g.:

```csharp
private void SomeFunction()
{
   object LocalFunction() => new object();
   this. DoSomethingElse(LocalFunction);
}
```

When invoking &quot;LocalFunction&quot; from whithin &quot;SomeFunction&quot;, IDE0009 is triggered which is not possible to be fixed.

---
#### Visual Studio Feedback System on 8/15/2019, 09:16 PM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

#### Visual Studio Feedback System on 8/15/2019, 10:51 PM: 

<p>This issue is currently being investigated. Our team will get back to you if either more information is needed, a workaround is available, or the issue is resolved.</p>

---
### Original Solutions
(no solutions)

**Comment by wellmonge** [comment]
@vsfeedback how can we reproduce this scenario?

**Comment by sharwell** [comment]
@wellmonge Set the various "qualification" settings to `true:warning` and use the code from the sample. See here for the basic form for these configuration entries:

https://github.com/dotnet/roslyn/blob/71e5206c0d8f8baf4e88f29efae1f39fb0bbcddb/.editorconfig#L46-L50
